src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3549,6 +3549,9 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
             fieldsByName.Free();
             memberNames.Free();
 
+            // Synthesizing non-readonly properties in struct would require changing readonly logic for PrintMembers method synthesis
+            Debug.Assert(isRecordClass || !members.Any(m => m is PropertySymbol { GetMethod.IsEffectivelyReadOnly : false }));
```

Comment:
```suggestion
            Debug.Assert(isRecordClass || !members.Any(m => m is PropertySymbol { GetMethod.IsEffectivelyReadOnly: false }));
```
nit: formatting

Reply:
For some reason, I'm getting formatting warning about this. Is it a problem with my environment? Or misconfiguration in `.editorconfig`
![image](https://user-images.githubusercontent.com/2550113/126837691-15bd427d-aed4-4e8e-9d01-12335f9bca0a.png)

Reply:
Looks like you are experiencing #54759. Let's just go with what the formatting analyzer demands for now.

Reply:
Looks like it, thanks for finding it. I'm fine with applying the change since it would be correct formatting after analyzer fix (it is also the style for patterns I see in the code), and current formatting with spaces on both sides looks strange :)

Reply:
The trouble with applying the change is that some of the CI legs will fail due to presence of warnings :/

Reply:
Oh, sorry, I'll revert it back.

Reply:
Reverted